### PR TITLE
Update auth for litellm proxy

### PR DESCRIPTION
**Pull request by pablodanswer**
## Description
Allows users to pass access token to litellm proxy set up

## How Has This Been Tested?
- [ ] Construct auth-enabled and non-auth-enabled LiteLLM proxy + test
- [ ] Construct other providers after testing

New test coming soon in sibling PR to validate this entire flow in the future.

## Accepted Risk
[Any know risks or failure modes to point out to reviewers]

## Related Issue(s)
[If applicable, link to the issue(s) this PR addresses]

## Checklist:
- [ ] All of the automated tests pass
- [ ] All PR comments are addressed and marked resolved
- [ ] If there are migrations, they have been rebased to latest main
- [ ] If there are new dependencies, they are added to the requirements
- [ ] If there are new environment variables, they are added to all of the deployment methods
- [ ] If there are new APIs that don't require auth, they are added to PUBLIC_ENDPOINT_SPECS
- [ ] Docker images build and basic functionalities work
- [ ] Author has done a final read through of the PR right before merge

### CODE REVIEW

File: web/src/app/admin/embeddings/pages/utils.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+//
+
+import { EmbeddingProvider } from "@/components/embedding/interfaces";
+
+// @router.delete("/delete-search-settings")
```
Comment (rkuo-danswer) [inline web/src/app/admin/embeddings/pages/utils.ts]: should we remove this commented code?

File: web/src/app/admin/embeddings/modals/ProviderCreationModal.tsx [hunk 2]
```diff
@@ -205,8 +220,8 @@ export function ProviderCreationModal({
                 ) : (
                   <TextFormField
                     name="api_key"
-                    label="API Key"
-                    placeholder="API Key"
+                    label={`API Key ${isProxy && "(for non-local deployments)"}`}
+                    placeholder={`API Key `}
```
Comment (rkuo-danswer) [inline web/src/app/admin/embeddings/modals/ProviderCreationModal.tsx]: Should there be a space on the end here? Are the backticks useful here?

**Review by rkuo-danswer — APPROVED** [review]
approved, some possible small cleanup in comments

File: web/src/app/admin/embeddings/pages/utils.ts (hunk 1 above)
Comment (pablonyx) [reply web/src/app/admin/embeddings/pages/utils.ts]: Good call!

File: web/src/app/admin/embeddings/modals/ProviderCreationModal.tsx (hunk 2 above)
Comment (pablonyx) [reply web/src/app/admin/embeddings/modals/ProviderCreationModal.tsx]: Good catch!

**Pull request merged**